### ISSUE TITLE
Improve support for .NET Standard

### ISSUE THREAD

**Issue by Perksey** [jbevain/cecil#623]
Currently, it's no picnic to output .NET Standard libs on Cecil. Granted, it is possible, but with a lot of added complexity (such as finding the ref `netstandard.dll` in the NuGet Package Cache manually). It'd be nice if Cecil had some way of handling this within the library, such as a `UseNetStandard` flag which will automatically map `System.Private.CoreLib` and `mscorlib` references to `netstandard`.

There's a number of ways this could be tackled in the library, but I want this issue to serve as a little discussion issue regarding the matter.

**Comment by jbevain** [comment]
Hey,

Thanks for filing thing. I'm not opposed to making things easier to deal with `netstandard`.

I'm however not sure I understand fully the implications or the requirements. Could you provide more details as to what you're thinking about? Thanks!

**Comment by Perksey** [comment]
So I've been working with Cecil for my upcoming library, most of the issues i was facing were silly mistakes on my part but the main thing i'm after is better support for .NET Standard within the TypeSystem, as TypeSystem.* still seems to add a reference to System.Private.CoreLib. If there's not already a ".NET Standard mode" or flag or something, one should be added to make netstandard and cecil friends again.

**Comment by bartico6** [comment]
It would seem you cannot actually set the assembly to target, say, .NET Standard or .NET Core even since TargetRuntime seems to define only .NET Framework versions ( as seen in https://github.com/jbevain/cecil/blob/master/Mono.Cecil/TargetRuntime.cs )

Are there any plans or ideas how to approach this issue? It'd definitely help, especially with "porting" binaries between .NET platforms, given enough manual patching beforehand (remove unsupported apis etc)

**Comment by Perksey** [comment]
netstandard.dll on all platforms that implement the .NET Standard is simply
one big library containing type forwards to other relevant libraries.
Perhaps Cecil could check if netstandard.dll contains a type forward for a
corelib type when importing, and if it does use netstandard instead of a
platform specific library

**Comment by jbevain** [comment]
As stated above, I'm not opposed to providing better supports for .NET Standard scenarios, but I'm still not sure I understand the issue.

If you want to help me help you, I'd love to see code that's not working or code that shows the extent of the hacks you need to do so support .NET Standard.

**Comment by Perksey** [comment]
I switched to System.Runtime.Metadata so this no longer concerns me, however I’ll leave it open for tracking and so @bartico6 can expand on their situation

**Comment by bartico6** [comment]
Hey,

As of a few days ago I've worked around the issue and it's a case of "I did a lot of things and a combination of them eventually worked" so I'm not entirely sure what was and what wasn't Mono.Cecil's fault. I'll look into preparing a real test case (as the issue I was having is resolved) and get back to you.  
  
However, for now, I think @Perksey can close the issue as my "experiment" with getting a .NET Framework 4.0 game to run on .NET Core has largely succeeded, but again, through a combination of things I'm unsure which had anything to do with Mono.Cecil, or with ILRepack (which is another part of the project that gave me headaches).  
When I have something concrete I'll definitely open an issue if this one is closed by that time!  
And of course thanks for offering help :)
